pages/api/roles/world/index.ts

```diff
@@ -1,5 +1,105 @@
+import { PrismaClient, Prisma, RoleType } from '@prisma/client';
 import type { NextApiRequest, NextApiResponse } from 'next';
+import { ApiResponse } from '../../../../types/apiResponse';
+import { WorldRoleListData, WorldRoleQueryParams } from '../../../../types/role';
+import { getNonEmptyArrayOrUndefined, combineDefinedArrays, buildFrequencyMap } from '../../../../utils/arrays';
+import { withVerifiedUser } from '../../../../utils/auth/jwtHelpers';
+import {
+  makeCompanyNameFilters,
+  makeRoleTitleFilters,
+  makeRoleYearFilters,
+  makeRoleTypeFilters,
+} from '../../../../utils/filters/filterHelpers';
+import {
+  convertQueryParamToStringArray,
+  createJsonResponse,
+  HttpMethod,
+  HttpStatus,
+  rejectHttpMethod,
+} from '../../../../utils/http/httpHelpers';
+import { withPrismaErrorHandling } from '../../../../utils/prisma/prismaHelpers';
+import { splitByWhitespaces, splitByCommaRemovingWhitespacesAround } from '../../../../utils/strings/formatters';
 
-export default function handler(req: NextApiRequest, res: NextApiResponse) {
-  res.status(200).end(`Request sent to ${req.url}\n`);
+const prisma = new PrismaClient();
+
+async function handler(userId: string, req: NextApiRequest, res: NextApiResponse) {
+  const method = req.method;
+  switch (method) {
+    case HttpMethod.GET:
+      return handleGet(userId, req, res);
+    default:
+      return rejectHttpMethod(res, method);
+  }
+}
+
+async function handleGet(userId: string, req: NextApiRequest, res: NextApiResponse<ApiResponse<WorldRoleListData[]>>) {
```

Comment:
Can you help me verify that this https://github.com/sivayogasubramanian/caza/pull/45/commits/ee9528844a5519e421164ef8285fef11edf2c46d is what you mean by changing the handler method signature? We will still need the first string parameter (albeit unused) since we are using `withVerifiedUser` here.

Reply:
Actually the current main has a new middleware `withVerified`, could you merge and use that?